src/ucs/arch/x86_64/cpu.c

```diff
@@ -462,13 +462,15 @@ static size_t ucs_cpu_memcpy_thresh(size_t user_val, size_t auto_val)
 void ucs_cpu_init()
 {
 #if ENABLE_BUILTIN_MEMCPY
+    /* use L3/2 (half of L3) memcpy threshold - this is the most performance value */
```

Comment:
maybe just put the whole code block under "if (l3_cache > 0) .. else { ...=INF}"

Reply:
should we respect user's values in case if L3 is not available?

Reply:
yes
if user provides values they are used
if user sets auto, then: if l3 cache - use it, otherwise - disable

consider updating ucs_cpu_memcpy_thresh function..